src/reaper_osara.cpp

```diff
@@ -402,16 +402,41 @@ string formatCursorPosition(TimeFormat format, FormatTimeCacheRequest cache) {
 	return formatTime(GetCursorPosition(), format, false, cache);
 }
 
-const char* formatFolderState(int state, bool reportTrack=true) {
+const string formatFolderState(MediaTrack* track) {
+	ostringstream s;
+	int state = (int)GetMediaTrackInfo_Value(track, "I_FOLDERDEPTH");
 	if (state == 0) {
 		// Translators: A track which isn't a folder.
-		return reportTrack ? translate("track") : nullptr;
+		s << translate("track");
+	} else if (state == 1 && GetMediaTrackInfo_Value(track, "P_PARTRACK")) {
+		// Translators: A track which is a nested folder.
+		s << translate("nested folder");
 	} else if (state == 1) {
 		// Translators: A track which is a folder.
-		return translate("folder");
+		s << translate("folder");
+	} else {
+		// Translators: A track which ends its folder.
+		s << translate("end of folder");
+		int trackNum = (int)GetMediaTrackInfo_Value(track, "IP_TRACKNUMBER");
+		// find the folder being ended by this track
+		MediaTrack* folderTrack;
+		int folderTrackNum{0};
+		for(folderTrackNum = trackNum -1; folderTrackNum>0; folderTrackNum--){
+			folderTrack= GetTrack(nullptr, folderTrackNum-1); // getTrack is 0 based
```

Comment:
Is there a reason we don't use GetParentTrack here? As it is now, we have to walk backwards through all the intervening tracks. If we use GetParentTrack, we can get the parent's track number after we find it. I realise we still need the folder depth, but that's fine; we can get that after calling GetParentTrack.

Reply:
There wasn't any good reason. I changed it to use GetParentTrack.